### PR TITLE
feat(metrics): Metrics extraction config, custom tags [INGEST-421] [INGEST-334] [INGEST-336]

### PR DESCRIPTION
**Pull request by untitaker**
Counterpart to https://github.com/getsentry/sentry/pull/30189

### CODE REVIEW

File: relay-server/src/actors/envelopes.rs [hunk 1]
```diff
@@ -1612,13 +1521,14 @@ impl EnvelopeProcessor {
         &self,
         state: &mut ProcessEnvelopeState,
     ) -> Result<(), ProcessingError> {
-        if !state.project_state.has_feature(Feature::MetricsExtraction) {
-            return Ok(());
-        }
+        let config = match state.project_state.config.transaction_metrics {
+            Some(ErrorBoundary::Ok(ref config)) => config,
+            None | Some(ErrorBoundary::Err(_)) => return Ok(()),
```
Comment (jjbayer) [inline relay-server/src/actors/envelopes.rs]: nit: Is this equivalent to `_ => return Ok(())`?

File: relay-server/src/metrics_extraction/transactions.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    if event.ty.value() != Some(&EventType::Transaction) {
+        return;
+    }
+
+    if config.extract_metrics.is_empty() {
+        return;
+    }
+
+    let mut push_metric = move |metric: Metric| {
+        if config.extract_metrics.contains(&metric.name) {
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions.rs:36]: nit: If the config check is the last thing we do for every metric, I would refactor to something like

```rust
extract_transaction_metrics(&event, &mut target); // no config!
target.retain(|x| config.extract_metrics.contains(x));
```

Although that opens the question how to handle tags. So probably this is fine the way it is.

Comment (untitaker) [reply relay-server/src/metrics_extraction/transactions.rs:36]: i want to early-return if there's no metrics to extract too, i think if i have the config handling in this function i can also unit-test it better

File: relay-server/src/actors/envelopes.rs (hunk 1 above)
Comment (untitaker) [reply relay-server/src/actors/envelopes.rs]: yes, i have the tendency to avoid `_` but it's not healthy in this case

**Review by jjbayer — APPROVED** [review]
There's an import error, and I assume the project config is missing from the relevant integration tests, but conceptually this looks good!

**Pull request merged**